homeassistant/components/kaleidescape/media_player.py

```diff
@@ -0,0 +1,136 @@
+"""Kaleidescape Media Player."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, cast
+
+from homeassistant.components.media_player import MediaPlayerEntity
+from homeassistant.components.media_player.const import (
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_STOP,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+)
+from homeassistant.util.dt import utcnow
+
+from .const import DOMAIN as KALEIDESCAPE_DOMAIN
+from .entity import KALEIDESCAPE_PLAYING_STATES, KaleidescapeEntity
+
+if TYPE_CHECKING:
+    from datetime import datetime
+
+    from kaleidescape import Device as KaleidescapeDevice
+
+    from homeassistant.config_entries import ConfigEntry
+    from homeassistant.core import HomeAssistant
+    from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+
+SUPPORTED_FEATURES = (
+    SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+    | SUPPORT_PLAY
+    | SUPPORT_PAUSE
+    | SUPPORT_STOP
+    | SUPPORT_NEXT_TRACK
+    | SUPPORT_PREVIOUS_TRACK
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the platform from a config entry."""
+    entities = [KaleidescapeMediaPlayer(hass.data[KALEIDESCAPE_DOMAIN][entry.entry_id])]
+    async_add_entities(entities)
+
+
+class KaleidescapeMediaPlayer(KaleidescapeEntity, MediaPlayerEntity):
+    """Representation of a Kaleidescape device."""
+
+    def __init__(self, device: KaleidescapeDevice) -> None:
+        """Initialize media player."""
+        super().__init__(device)
+        self._attr_supported_features = SUPPORTED_FEATURES
+
+    async def async_turn_on(self) -> None:
+        """Send leave standby command."""
+        await self._device.leave_standby()
+
+    async def async_turn_off(self) -> None:
+        """Send enter standby command."""
+        await self._device.enter_standby()
+
+    async def async_media_pause(self) -> None:
+        """Send pause command."""
+        await self._device.pause()
+
+    async def async_media_play(self) -> None:
+        """Send play command."""
+        await self._device.play()
+
+    async def async_media_stop(self) -> None:
+        """Send stop command."""
+        await self._device.stop()
+
+    async def async_media_next_track(self) -> None:
+        """Send track next command."""
+        await self._device.next()
+
+    async def async_media_previous_track(self) -> None:
+        """Send track previous command."""
+        await self._device.previous()
+
+    @property
+    def available(self) -> bool:
+        """Return if device is available."""
+        return cast(bool, self._device.is_connected)
```

Comment:
If your library implements typing and PEP561 you can avoid the casts

https://www.python.org/dev/peps/pep-0561/

Reply:
Ah thanks, that explains my total confusion over the mypy errors about `Any`. I have now implemented PEP561 in the library, and removed the casts here. mypy is happy again.